dist/js/components/Product.js

```diff
@@ -0,0 +1,218 @@
+      }
+      
+      thisProduct.dom.cartButton.addEventListener('click', function(event){
+        event.preventDefault();
+        thisProduct.processOrder();
+        thisProduct.addToCart();
+      });
+    }
+
+    processOrder() {
+      const thisProduct = this;
+    
+      // covert form to object structure e.g. { sauce: ['tomato'], toppings: ['olives', 'redPeppers']}
+      const formData = utils.serializeFormToObject(thisProduct.form);
+    
+      // set price to default price
+      let price = thisProduct.data.price;
+    
+      // for every category (param)...
+      for(let paramId in thisProduct.data.params) {
+        // determine param value, e.g. paramId = 'toppings', param = { label: 'Toppings', type: 'checkboxes'... }
+        const param = thisProduct.data.params[paramId];
+        //console.log(paramId, param);
+    
+        // for every option in this category
+        for(let optionId in param.options) {
+          // determine option value, e.g. optionId = 'olives', option = { label: 'Olives', price: 2, default: true }
+          const option = param.options[optionId];
+          const optionSelected = formData[paramId] && formData[paramId].includes(optionId);
+          //console.log(optionId, option);
+          //console.log(paramId, optionSelected)
+          // check if there is param with a name of paramId in formData and if it includes optionId
+        if (optionSelected) {
+          // check if the option is not default
+          if (!option.default == true) {
+            // add option price to price variable
+            price += option.price;
+          }
+        } else {
+          // check if the option is default
+          if (option.default == true) {
+            // reduce price variable
+            price -= option.price;
+          }
+        }
+          // add image 
+        const optionImage = thisProduct.dom.imageWrapper.querySelector('.' + paramId + '-' + optionId);
+        
+        if (optionImage) {
+          if (optionSelected) {
+            optionImage.classList.add(classNames.menuProduct.imageVisible);
+          }
+          else {
+            optionImage.classList.remove(classNames.menuProduct.imageVisible);
+          }
+        }
+      }
+    }
+    //single price
+    thisProduct.priceSingle = price;
+
+    // multiply price by amount
+    price *= thisProduct.amountWidget.value;
+
+    // update calculated price in the HTML
+    thisProduct.dom.priceElem.innerHTML = price;
+    }
+
+    initAmountWidget(){
+      const thisProduct = this;
+
+      thisProduct.amountWidget = new AmountWidget(thisProduct.dom.amountWidgetElem);
+      thisProduct.dom.amountWidgetElem.addEventListener('updated',function(){
+        thisProduct.processOrder();
+      });
+    }
+
+    addToCart(){
+      const thisProduct = this;
+      //app.cart.add(thisProduct.prepareCartProduct());
+
+      const event = new CustomEvent('add-to-cart', {
+        bubbles: true,
+        details: {
```

Comment:
Tutaj powinno być `detail` i powinno działać
```suggestion
        detail: {
```